### PR TITLE
Issue #38: decouple drupal download from drupal install

### PR DESCRIPTION
**Pull request by mradcliffe**
I might as well open this for review so that I can make changes or figure out what tests are needed.

I just rebased, but for some reason had to force push to my remote.

### CODE REVIEW

File: functions/drupal.sh [hunk 1]
```diff
@@ -23,12 +23,22 @@ function drupal_ti_ensure_drupal() {
...
+#
+# Ensures that Drupal is installed.
+#
+function drupal_ti_ensure_drupal_installed() {
+	drupal_ti_ensure_drupal
+
+	# Create database and install Drupal.
+	mysql -e "create database $DRUPAL_TI_DB"
+
+	cd drupal
```
Comment (LionsAd) [inline functions/drupal.sh]: This should be:

```
cd "$DRUPAL_TI_DRUPAL_BASE"
```

File: functions/drupal.sh [hunk 2]
```diff
@@ -23,12 +23,22 @@ function drupal_ti_ensure_drupal() {
...
 	cd "$DRUPAL_TI_DRUPAL_BASE"
 
+	drupal_ti_download_drupal
+}
+
+#
+# Ensures that Drupal is installed.
+#
+function drupal_ti_ensure_drupal_installed() {
+	drupal_ti_ensure_drupal
```
Comment (LionsAd) [inline functions/drupal.sh]: This function is not re-entrant anymore, as is not the other one.

File: runners/behat/before_script.sh [hunk 3]
```diff
@@ -4,7 +4,7 @@
 set -e $DRUPAL_TI_DEBUG
 
 # Ensure the right Drupal version is installed.
-drupal_ti_ensure_drupal
+drupal_ti_ensure_drupal_installed
```
Comment (LionsAd) [inline runners/behat/before_script.sh]: I like that a lot! :+1:

Comment (LionsAd) [inline runners/behat/before_script.sh]: However I think I would rather do it the other way round for BC reasons,

e.g. drupal_ti_ensure_drupal_downloaded + drupal_ti_ensure_drupal

File: runners/phpunit-core/before_script.sh [hunk 4]
```diff
@@ -3,8 +3,11 @@
...
 set -e $DRUPAL_TI_DEBUG
 
-# Ensure the right Drupal version is installed.
+# Ensure the right Drupal version is downloaded.
 drupal_ti_ensure_drupal
 
-# Ensure the module is linked into the code base and enabled.
-drupal_ti_ensure_module
+# Change directory to Drupal core directory.
+cd drupal
```
Comment (LionsAd) [inline runners/phpunit-core/before_script.sh]: This should not be necessary for BC reasons.

**Comment by mradcliffe** [comment]
I have not had a chance to work on this lately. Sorry.

**Comment by yanniboi** [comment]
Hello @mradcliffe and @LionsAd, I have had a chance to look at this however the first thing I did was rebase issue-38 on master which needed some merge conflicts solving, so I am not sure what the best way to contribute to this PR is. 

I can create a PR into mradcliffe:issue-38 but that will be messy to review, so I have also created a PR from yanniboi:issue-38 to yanniboi:master (matching LionsAd:master) on my fork which makes the overall review and testing easier (maybe...) here: https://github.com/yanniboi/drupal_ti/pull/1

What I have mainly done is, rather than changing the behat and simpletest runners to use a different function to install drupal, I have made it so 'drupal_ti_ensure_drupal' still downloads and installs drupal, but it downloads drupal by calling another re-entrant function 'drupal_ti_ensure_drupal_download' which can be called by phpunit-core rather than 'drupal_ti_ensure_drupal'. This means no changes are needed to behat and simpletest.

Let me know what the best way to get this into the current PR is or if it is easier to create a new PR?

**Comment by yanniboi** [comment]
I have spoken to @mradcliffe on IRC and he is happy for me to start a new PR from my fork. Will post a link to the new PR here.

**Comment by mradcliffe** [comment]
Closing in favor of other pull request.

**Pull request closed without merging**